### PR TITLE
build: docker-compose update for development

### PR DESCRIPTION
**Pull request by rjdp**
Mount local files into container, gunicorn autoreloads on file changes
add celerybeat-schedule.db to .gitignore
mound docker.sock to silence celery worker errors in log

## Iteration 2
- Use autoreload clery-worker scripts for dev docker-compose
- Use `DEV` build arg for Dev build ( installs dev dependencies from  `requirements/dev.txt` )
- Add  dev docker-compose file for third-party

### CODE REVIEW

**Review by Samyak2 — CHANGES_REQUESTED** [review]
Would be nice to have auto-reload for celery workers and scheduler too. Refer:
- https://github.com/chaos-genius/chaos_genius/blob/main/scripts/celery/dev_worker-alerts.sh
- https://github.com/chaos-genius/chaos_genius/blob/main/scripts/celery/dev_worker-analytics.sh
- https://github.com/chaos-genius/chaos_genius/blob/main/scripts/celery/dev_scheduler.sh

File: Dockerfile [hunk 1]
```diff
@@ -7,6 +7,8 @@ RUN apt-get update \
 
 COPY requirements /requirements
 
-RUN pip install -r /requirements/prod.txt --no-cache-dir
+ARG DEV
+
+RUN pip install -r /requirements/prod.txt ${DEV:+-r /requirements/dev.txt} --no-cache-dir
```
Comment (Samyak2) [inline Dockerfile:12]: This is nice. Tested it locally and it works as expected.

**Review by Samyak2 — CHANGES_REQUESTED** [review]
Looks good overall. A few changes in the newly added compose files may be required.

File: docker-compose.thirdparty.dev.yaml [hunk 2]
```diff
@@ -0,0 +1,429 @@
+version: "3.7"
```
[the remark this answers is not in the document]
Comment (rjdp) [reply docker-compose.thirdparty.dev.yaml]: Done

File: docker-compose.thirdparty.dev.yaml [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+    command: sh setup/run-backend-docker.sh --reload
+    volumes:
+      - ./:/usr/src/app/
+      - /var/run/docker.sock:/var/run/docker.sock
+    depends_on:
+      - chaosgenius-db
+    environment:
+      - *chaosgenius-version
+      - IN_DOCKER=True
+      - AIRBYTE_ENABLED=${AIRBYTE_ENABLED}
```
[the remark this answers is not in the document]
Comment (rjdp) [reply docker-compose.thirdparty.dev.yaml]: done

File: docker-compose.dev-thirdparty.yml [hunk 4]
```diff
@@ -0,0 +1,429 @@
+version: "3.7"
+#https://github.com/compose-spec/compose-spec/blob/master/spec.md#using-extensions-as-fragments
+x-logging: &default-logging
+  options:
+    max-size: "100m"
+    max-file: "5"
+  driver: json-file
+x-version:
+  &chaosgenius-version
+  CHAOSGENIUS_VERSION_POSTFIX=stable-thirdparty-compose
```
Comment (Samyak2) [inline docker-compose.dev-thirdparty.yml:10]: This should be named `dev-thirdparty-compose`
```suggestion
  CHAOSGENIUS_VERSION_POSTFIX=dev-thirdparty-compose
```

File: docker-compose.dev-thirdparty.yml [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+      - REACT_APP_BASE_URL=${REACT_APP_BASE_URL}
+      - REACT_APP_DISABLE_TELEMETRY=${REACT_APP_DISABLE_TELEMETRY}
+      - REACT_APP_EVENT_ALERT=${REACT_APP_EVENT_ALERT}
+    depends_on:
+      - chaosgenius-server
+
+  chaosgenius-db:
+    container_name: chaosgenius-db
+    image: postgres:13
+    restart: always
```
Comment (Samyak2) [inline docker-compose.dev-thirdparty.yml:119]: We don't have restart policies for the other services, so we should remove it from this one too

File: docker-compose.dev-thirdparty.yml [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+    healthcheck:
+      test: [ "CMD", "pg_isready", "-q", "-d", "chaosgenius_data", "-U", "postgres" ]
+      timeout: 45s
+      interval: 10s
+      retries: 10
+    restart: always
```
Comment (Samyak2) [inline docker-compose.dev-thirdparty.yml:125]: Here too

**Review by Samyak2 — COMMENTED** [review]
Noticed a couple of more things now

File: docker-compose.dev-thirdparty.yml (hunk 5 above)
Comment (rjdp) [reply docker-compose.dev-thirdparty.yml:119]: @Samyak2  what do think about restart on-failure for db and redis ? or still default no will be better ?

File: docker-compose.dev-thirdparty.yml (hunk 6 above)
Comment (rjdp) [reply docker-compose.dev-thirdparty.yml:125]: cool, btw restart policies on airbyte dockers are fine?

File: docker-compose.dev-thirdparty.yml (hunk 5 above)
Comment (rjdp) [reply docker-compose.dev-thirdparty.yml:119]: going for on-failure for all "dev" compose services

**Review by Samyak2 — APPROVED** [review]
Looks good!

**Pull request merged**